awscli/autoprompt/factory.py

```diff
@@ -191,26 +243,51 @@ def create_layout(self, on_input_buffer_text_changed=None,
                 self.create_input_buffer(on_input_buffer_text_changed)
             input_buffer_container = \
                 self.create_input_buffer_container(input_buffer)
-        if doc_window is None:
-            doc_buffer = self.create_doc_buffer()
-            doc_window = self.create_doc_window(doc_buffer)
         if search_field is None:
             search_field = SearchToolbar()
+        if doc_window is None:
+            doc_buffer = self.create_doc_buffer()
+            doc_window = self.create_doc_window(doc_buffer, search_field)
         if bottom_toolbar_container is None:
             bottom_toolbar_buffer = self.create_bottom_toolbar_buffer()
             bottom_toolbar_container = \
                 self.create_bottom_toolbar_container(bottom_toolbar_buffer)
+        help_doc_buffer = self.create_help_buffer(
+            self._help_text.full_doc_window_key_binding_text
+        )
+        help_doc_window = self.create_help_window(
+            help_doc_buffer, title='Doc panel help')
+        help_input_buffer = self.create_help_buffer(
+            self._help_text.full_input_buffer_key_binding_text
+        )
```

Comment:
Is there a particular reason we are introducing a new pattern for a creating a container for every new type of help panel? From the looks of it the toolbar just uses a single container and we update the input buffer based on the keypresses. I sort of like that approach because it makes following the layout of the app easier especially if we add more help panels for more views.

Reply:
Yes. Because title in frame is immutable.

Reply:
That's fair. I think my biggest feedback then is let's make a helper method, factory, or even a new prompt-toolkit class (e.g. implement the `__pt_container__` method) to encompass all of these containers as it makes it hard to understand this top-level view of the app structure. Ideally this utility would just return a single container with all of the full help windows and we can just add that container to the top-level view.